### ISSUE TITLE
qa: Cancel running queries

### ISSUE THREAD

**Issue by vivekmenezes** [cockroachdb/cockroach#21878]

**Comment by asubiotto** [comment]
https://github.com/cockroachdb/cockroach/issues/19951 was created and fixed as a result of this work.
https://github.com/cockroachdb/cockroach/issues/19797 is a user issue that still needs to be resolved but I'm not sure it's related to query cancellation.

I tested the query cancellation on a scalefactor 10 database by cancelling queries like `SELECT COUNT(*) FROM tpch.lineitem` and `SELECT l_shipmode, AVG(l_extendedprice) FROM tpch.lineitem GROUP BY l_shipmode`.

One thing I noticed is that non-distsql queries respond to cancellation much quicker than distsql queries. There might be some investigation/work to do there. Filed https://github.com/cockroachdb/cockroach/issues/22650.

Otherwise, query cancellation seems to be behaving fine.
